gen3-client/logs/failed.go

```diff
@@ -0,0 +1,125 @@
+	if err != nil {
+		failedLogFile.Close()
+		log.Fatal("Error occurred when opening file \"" + failedLogFilename + "\": " + err.Error())
+	}
+	log.Println("Local failed log file \"" + failedLogFilename + "\" has opened")
+
+	failedLogFileMap = make(map[string]commonUtils.RetryObject)
+}
+
+func LoadFailedLogFile(filePath string) {
+	file, err := os.OpenFile(filePath, os.O_RDONLY, 0766)
+	if err != nil {
+		file.Close()
+		failedLogFile.Close()
+		log.Fatal("Error occurred when opening file \"" + file.Name() + "\": " + err.Error())
+	}
+	fi, err := file.Stat()
+	if err != nil {
+		file.Close()
+		failedLogFile.Close()
+		log.Fatal("Error occurred when opening file \"" + file.Name() + "\": " + err.Error())
+	}
+	log.Println("Failed log file \"" + file.Name() + "\" has been opened for read")
+
+	if fi.Size() > 0 {
+		var tempRetryObjectSlice []commonUtils.RetryObject
+		data, err := ioutil.ReadAll(file)
+		if err != nil {
+			file.Close()
+			failedLogFile.Close()
+			log.Fatal("Error occurred when reading from file \"" + file.Name() + "\": " + err.Error())
+		}
+
+		err = json.Unmarshal(data, &tempRetryObjectSlice)
+		if err != nil {
+			file.Close()
+			failedLogFile.Close()
+			log.Fatal("Error occurred when unmarshaling from JSON objects: " + err.Error())
+		}
+
+		for _, ro := range tempRetryObjectSlice {
+			failedLogFileMap[ro.FilePath] = ro
+		}
+	}
+}
+
+func IsFailedLogMapEmpty() bool {
+	return len(failedLogFileMap) == 0
+}
+
+func GetFailedLogMap() map[string]commonUtils.RetryObject {
+	return failedLogFileMap
+}
+
+func AddToFailedLogMap(filePath string, guid string, presignedUrl string, retryCount int, isMuted bool) {
+	failedLogLock.Lock()
+	defer failedLogLock.Unlock()
+	failedLogFileMap[filePath] = commonUtils.RetryObject{FilePath: filePath, GUID: guid, PresignedURL: presignedUrl, RetryCount: retryCount}
+	if !isMuted {
+		log.Printf("Failed file entry added for %s\n", filePath)
+	}
+}
+
+func DeleteFromFailedLogMap(filePath string, isMuted bool) {
+	failedLogLock.Lock()
+	defer failedLogLock.Unlock()
+	delete(failedLogFileMap, filePath)
+	if !isMuted {
+		log.Printf("Failed file entry deleted for %s\n", filePath)
+	}
+}
+
+func WriteToFailedLog(isMuted bool) {
+	failedLogLock.Lock()
+	defer failedLogLock.Unlock()
+	var tempSlice []commonUtils.RetryObject
+	for _, v := range failedLogFileMap {
+		tempSlice = append(tempSlice, v)
+	}
+	if len(tempSlice) == 0 {
+		tempSlice = []commonUtils.RetryObject{}
+	}
+	jsonData, err := json.Marshal(tempSlice)
+	if err != nil {
+		failedLogFile.Close()
+		log.Fatal("Error occurred when marshaling to JSON objects: " + err.Error())
+	}
+	err = failedLogFile.Truncate(0)
+	_, err = failedLogFile.WriteAt(jsonData, 0)
+	if err != nil {
+		failedLogFile.Close()
+		log.Fatal("Error occurred when writing to file \"" + failedLogFilename + "\": " + err.Error())
+	}
+	if !isMuted {
+		log.Println("Local failed log file updated")
```

Comment:
I find the phrase "Local failed log file" confusing. Is this message necessary? I think the print you have above, "Failed file entry added for logFilePath", is more helpful and maybe sufficient

Reply:
Yeah I think this is redundant and can be get rid of. But I kind want to keep the message from opening/closing these log files since these are like some major waypoints during execution